### PR TITLE
Parallell wildcard elimination

### LINKED ISSUE #2: Improve wildcard elimination speed by adding concurrency to the process
Currently wildcard elimination is done in a single threaded way, it doesnt takes much time however it would be nice to have multi-threaded resolving. Need to benchmark and see.

### PR DESCRIPTION
**Pull request by Mzack9999**
Closes #2

### CODE REVIEW

File: pkg/massdns/process.go [hunk 1]
```diff
@@ -55,57 +56,85 @@ func (c *Client) Process() error {
...
+				// // If the same ip has been found more than 5 times, perform wildcard detection
+				// // on it now, if an IP is found in the wildcard we add it to the wildcard map
+				// // so that further runs don't require such filtering again.
+				if ipm.Counter >= 5 && !ipm.Validated {
+					for host := range ipm.Hostnames {
+						wildcard, ips := c.wildcardResolver.LookupHost(host)
+						if wildcard {
+							for ip := range ips {
+								c.wildcardIPMap[ip] = struct{}{}
+							}
```
Comment (Ice3man543) [inline pkg/massdns/process.go]: Needs a lock here. Concurrent manipulation will lead to panics imo

File: pkg/runner/runner.go [hunk 2]
```diff
@@ -162,14 +162,15 @@ func (r *Runner) processSubdomains() {
...
-		Threads:       r.options.Threads,
-		InputFile:     inputFile,
-		ResolversFile: r.options.ResolversFile,
-		TempDir:       r.tempDir,
-		OutputFile:    r.options.Output,
+		Domain:           r.options.Domain,
+		Retries:          r.options.Retries,
+		MassdnsPath:      r.options.MassdnsPath,
+		Threads:          r.options.Threads,
+		WildcardsThreads: r.options.WildcardThreads,
```
Comment (Ice3man543) [inline pkg/runner/runner.go]: Modify the readme to specify what is wildcard threads and how to use it.

File: pkg/massdns/process.go (hunk 1 above)
Comment (Mzack9999) [reply pkg/massdns/process.go]: ouch forgot, done!

File: pkg/runner/runner.go (hunk 2 above)
Comment (Mzack9999) [reply pkg/runner/runner.go]: done

**Comment by Ice3man543** [comment]
@Mzack9999 i tried to benchmark it and found no significant differences in performance and speed with single threaded and multi-threaded versions. Can you benchmark it and see if it makes a difference?

**Comment by Mzack9999** [comment]
Benchmark with a `3Gb` massdns output.

Multi-Thread (equal to `-wt 25`):
```
$ shuffledns -wildcard-only -massdns-file massdns.output.txt
real	3m30.186s
user	0m55.108s
sys	0m16.860s
```

Single-Thead (benchmark done also for `master` branch - wildcard elaboration part)
```
$ shuffledns -wildcard-only -massdns-file massdns.output.txt -wt 1
real	31m3.663s
user	0m55.808s
sys	0m17.636s
```

**Comment by Ice3man543** [comment]
@Mzack9999 I reworked the flags reducing the two flags to a single one, also simplified some of the code removing channels for sempahore based concurrency control. Can you check out my latest commit and see how it compares to the older one?

**Comment by Mzack9999** [comment]
@Ice3man543 The code looks good to me, I updated the readme that contained old flags

**Comment by Ice3man543** [comment]
@mzack9999, one more thing - can you run the test case to see if this one offers some speed difference as I removed worker pool.

**Comment by Mzack9999** [comment]
@Ice3man543 Everything works correctly, feel free to merge upon your final review

**Comment by Ice3man543** [comment]
@Mzack9999 this crashes in my tests. Any ideas?

**Pull request merged**